vulcanai2/tests/models/test_cnn.py

```diff
@@ -1,114 +1,140 @@
+        class_output = conv3D_net_class.metrics.extract_class_labels(
             in_matrix=raw_output)
         assert np.any(~np.isnan(class_output))
         assert np.any(~np.isnan(raw_output))
 
-    def test_freeze_class(self, cnn_class):
+    def test_freeze_class(self, conv3D_net_class):
         """Test class network freezing."""
-        cnn_class.freeze(apply_inputs=False)
-        for params in cnn_class.network.parameters():
+        conv3D_net_class.freeze(apply_inputs=False)
+        for params in conv3D_net_class.network.parameters():
             assert params.requires_grad is False
 
-    def test_unfreeze_class(self, cnn_class):
+    def test_unfreeze_class(self, conv3D_net_class):
         """Test class network unfreezing."""
-        cnn_class.freeze(apply_inputs=False)
-        cnn_class.unfreeze(apply_inputs=False)
-        for params in cnn_class.network.parameters():
+        conv3D_net_class.freeze(apply_inputs=False)
+        conv3D_net_class.unfreeze(apply_inputs=False)
+        for params in conv3D_net_class.network.parameters():
             assert params.requires_grad is True
 
-    def test_freeze_noclass(self, cnn_noclass):
+    def test_freeze_noclass(self, conv3D_net):
         """Test intermediate network freezing."""
-        cnn_noclass.freeze(apply_inputs=False)
-        for params in cnn_noclass.network.parameters():
+        conv3D_net.freeze(apply_inputs=False)
+        for params in conv3D_net.network.parameters():
             assert params.requires_grad is False
 
-    def test_unfreeze_noclass(self, cnn_noclass):
+    def test_unfreeze_noclass(self, conv3D_net):
         """Test intermediate network unfreezing."""
-        cnn_noclass.freeze(apply_inputs=False)
-        cnn_noclass.unfreeze(apply_inputs=False)
-        for params in cnn_noclass.network.parameters():
+        conv3D_net.freeze(apply_inputs=False)
+        conv3D_net.unfreeze(apply_inputs=False)
+        for params in conv3D_net.network.parameters():
             assert params.requires_grad is True
 
-    def test_add_input_network(self, cnn_noclass, cnn_class):
-        """Test add input Network functionality."""
-        cnn_class.add_input_network(cnn_noclass)
-        assert cnn_class.input_networks[cnn_noclass.name] is cnn_noclass
-        assert cnn_class.in_dim == cnn_noclass.out_dim
+    def test_fit_multi_input(self, multi_input_cnn,
+                             multi_input_cnn_data):
+        """Test for fit function"""
+        test_train = Subset(multi_input_cnn_data, 
+                            range(len(multi_input_cnn_data)//2))
+        test_val = Subset(multi_input_cnn_data, 
+                          range(len(multi_input_cnn_data)//2, 
+                                    len(multi_input_cnn_data)))
+        test_train_loader = DataLoader(test_train, batch_size=2)
+        test_val_loader = DataLoader(test_val, batch_size=2)
+        multi_input_cnn.fit(test_train_loader, test_val_loader, 1)
```

Comment:
@priyatharsan would you be able to assert that the weights (I.e. parameters) have changed? you can probably compare parts of the `state_dict`

Reply:
sure I will take a look into it tonight.